addons/account/models/account_move.py

```diff
@@ -2342,6 +2342,8 @@ def write(self, vals):
             for move in self:
                 if 'tax_totals' in vals:
                     super(AccountMove, move).write({'tax_totals': vals['tax_totals']})
+        if 'journal_id' in vals:
+            move.line_ids._check_constrains_account_id_journal_id()
```

Comment:
@agr-odoo we made a mistake here. :/
It should have been `self.line_ids._check_constrains_account_id_journal_id()`

Reply:
solved: https://github.com/odoo/odoo/pull/130020